File: dagger/app/src/main/java/com/dagger/mvp/model/RegisterModel.java
package com.dagger.mvp.model;

import android.support.annotation.NonNull;

import com.dagger.mvp.view.MVPView;

import dagger.Module;
import dagger.Provides;

/**
 * by y on 2017/5/31.
 */
@Module
public class RegisterModel {

    @NonNull
    private final MVPView mvpView;

    public RegisterModel(@NonNull MVPView mvpView) {
        this.mvpView = mvpView;
    }

    @Provides
    MVPView provideView() {
        return mvpView;
    }
}

File: UtilsSample/Dagger/src/main/java/sample/util/develop/android/dagger/mvp/model/RegisterModel.kt
package sample.util.develop.android.dagger.mvp.model

import dagger.Module
import dagger.Provides
import sample.util.develop.android.dagger.mvp.view.MVPView

/**
 * by y on 2017/5/31.
 */
@Module
class RegisterModel(private val mvpView: MVPView) {

    @Provides
    fun provideView() = mvpView
}
